overlord/devicestate/devicectx.go

```diff
@@ -104,6 +104,17 @@ func (dc groundDeviceContext) HasModeenv() bool {
 	return dc.model.Grade() != asserts.ModelGradeUnset
 }
 
+// IsCoreBoot is true when there are modes, or when there are not but
+// we are not in classic (UC16/18 case)
+func (d *groundDeviceContext) IsCoreBoot() bool {
+	return d.HasModeenv() || !d.Classic()
+}
+
+// IsClassicBoot is true for classic systems with classic initramfs
+func (d *groundDeviceContext) IsClassicBoot() bool {
+	return !d.IsCoreBoot()
```

Comment:
This expands to
```go
!(d.HasModeenv() || !d.Classic())
```
which is:
```go
(!d.HasModeenv() && d.Classic())
```
that is a classic system with no run mode set. I'm a bit ignorant of the boot process details (I guess we don't use run modes on classic? Or do we have the "install" mode there too?), so it's likely that this is correct, but I just wanted to double check with you. I wonder if the function docstring could be expanded a bit to clarify this (it mentions the initramfs, but no run mode).

Reply:
This PR is part of the changes for starting to use the UC initramfs on classic systems. So, we did not have modes for classic in the past, but now we will start to distinguish betwee classic with and without modes. I'll clarify a bit more in the docstring.

Reply:
It's clear now, thanks!